modin/pandas/dataframe.py

```diff
@@ -1387,19 +1391,25 @@ def merge(
                 raise ValueError("Cannot merge a Series without a name")
             else:
                 right = right.to_frame()
-        if not isinstance(right, DataFrame):
+        if not isinstance(right, (DataFrame, pandas.DataFrame)):
```

Comment:
we unconditionally converted `right` to pd.DataFrame above, why do we want to check for pandas.DataFrame here?